### PR TITLE
pull request template

### PR DESCRIPTION
**Pull request by sonichi**
add a template for PR

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+<!-- For example: "Closes #1234" -->
+
+## Checks
+
+- [ ] I've used [pre-commit](https://microsoft.github.io/FLAML/docs/Contribute#pre-commit) to lint the changes in this PR, or I've checked [lint with flake8](https://github.com/microsoft/FLAML/blob/816a82a1155b4de4705b21a615ccdff67c6da379/.github/workflows/python-package.yml#L54-L59) output is two 0s.
+- [ ] I've included any doc changes needed for https://microsoft.github.io/FLAML/.
+- [ ] I've made sure the tests are passing.
+- Testing Strategy
+   - [ ] Unit tests
+   - [ ] This PR is not tested :(
```
Comment (ZviBaratz) [inline .github/PULL_REQUEST_TEMPLATE.md:20]: What do you think about replacing this with:
- [ ] If relevant, added tests corresponding to the changes introduced in this PR.
- [ ] Tests run successfully.

**Review by ZviBaratz — APPROVED** [review]
Made a couple of minor suggestions, otherwise LGTM.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+<!-- Please give a short summary of the change and the problem this solves. -->
+
+## Related issue number
+
+<!-- For example: "Closes #1234" -->
+
+## Checks
+
+- [ ] I've used [pre-commit](https://microsoft.github.io/FLAML/docs/Contribute#pre-commit) to lint the changes in this PR, or I've checked [lint with flake8](https://github.com/microsoft/FLAML/blob/816a82a1155b4de4705b21a615ccdff67c6da379/.github/workflows/python-package.yml#L54-L59) output is two 0s.
```
[the remark this answers is not in the document]
Comment (sonichi) [reply .github/PULL_REQUEST_TEMPLATE.md:15]: I had trouble using pre-commit in Windows. So for those with the same trouble, I offer another option of fixing the lint issues manually.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (sonichi) [reply .github/PULL_REQUEST_TEMPLATE.md]: Updated.

**Review by int-chaos — APPROVED** [review]

**Review by qingyun-wu — APPROVED** [review]

**Review by liususan091219 — APPROVED** [review]

**Pull request merged**